### ISSUE TITLE
Request for advanced documentation on the text generation pipeline.

### ISSUE THREAD

**Issue by Jcwscience** [huggingface/transformers#14281]
# 🚀 Feature request
Detailed information on the various arguments that the pipeline accepts.

Explanation of the use cases described, eg. How to provide examples to prime the model for a task. 
## Motivation
I have hit a wall in several of my projects, and being relatively new to this field of computer science I can’t seem to get my foot in the door. Each model or example project I attempt to use seems to assume I already have certain knowledge. I am not sure what this knowledge is but since I find it hard to believe you could make serious progress with just the information available in the documentation, I feel there must be something I’m missing. If I’ve missed some obvious piece of information or documentation I apologize for the bother, but if not I could really use a bit of help. Thanks!

## Your contribution
I’m more than willing to help in any way I can within the realm of my abilities, however at the moment that realm is fairly small.

**Comment by LysandreJik** [comment]
Hello! The text generation pipeline takes as argument all of the `generate` method's argument, to tweak the generation. The documentation for that method is available here: https://huggingface.co/transformers/main_classes/model.html?highlight=generate#transformers.generation_utils.GenerationMixin.generate

Does that help?
cc @patrickvonplaten

**Comment by Jcwscience** [comment]
@LysandreJik There it is! That’s what I have been missing! Right in front of me. Thank you! I can’t believe I haven’t made it to that page before somehow.

**Comment by LysandreJik** [comment]
That's helpful feedback! The page should not be hard to find, we'll see what we can do to make it clearer. Anything in particular would have helped you find it? Where would you have expected to see a mention of it, for example?

**Comment by Jcwscience** [comment]
@LysandreJik Well the first place I landed when looking for info was the [Pipelines description page.](https://huggingface.co/transformers/main_classes/pipelines.html#transformers.TextGenerationPipeline) 

I think the confusing part was the fact that it has *some* parameters listed. If it didn’t have any I would probably have realized I needed to check somewhere else. But since it has a few I just assumed that was most of the info.

Maybe if there was just a link saying something like “more information” or “for a full list of parameters visit this page”? Just a thought. I know the models != the pipelines exactly, so I’m not sure what needs to be linked where for sure.
